realtek: dsa: rtl839x: Fix incorrect drop counter names

ecsv opened the pull request:
The counter names "CFM" and "VLAN_IGR_FLTR" were not separated by a comma in the `rtl839x_drop_cntr` array. As result, these two headers were merged to a single header "CFMVLAN_IGR_FLTR" and everything after that was shifted by one. The last name (for the 45th counter) was also not defined and was therefore accessing data outside the array.

Fixes: 1cfd45ae0bad ("realtek: Add debugfs support for RTL9300")

ecsv committed:
realtek: dsa: rtl839x: Fix incorrect drop counter names

The counter names "CFM" and "VLAN_IGR_FLTR" were not separated by a comma
in the `rtl839x_drop_cntr` array. As result, these two headers were merged
to a single header "CFMVLAN_IGR_FLTR" and everything after that was shifted
by one. The last name (for the 45th counter) was also not defined and was
therefore accessing data outside the array.

Fixes: 1cfd45ae0bad ("realtek: Add debugfs support for RTL9300")
Signed-off-by: Sven Eckelmann <sven@narfation.org>
Link: https://github.com/openwrt/openwrt/pull/20905
Signed-off-by: Robert Marko <robimarko@gmail.com>

hauke commented:
Please also add a `BUILD_BUG_ON` or other check to ensure we would detect this next time.

ecsv committed:
realtek: dsa: Build check size of drop counter names

The commit 1cfd45ae0bad ("realtek: Add debugfs support for RTL9300") caused
previously an out of bounds access on the array holding the names of drop
counters (and incorrect names in the output) fur RTL839x because of a
missing comma. To avoid such situation in the future, calculate the size of
the array during compilation. And to ensure that this count matches the
actual number of counters in HW, compare this number during compile time
with the expected value.

Suggested-by: Hauke Mehrtens <hauke@hauke-m.de>
Signed-off-by: Sven Eckelmann <sven@narfation.org>
Link: https://github.com/openwrt/openwrt/pull/20905
Signed-off-by: Robert Marko <robimarko@gmail.com>
